crates/protocol/src/quests/mod.rs

```diff
@@ -118,6 +118,51 @@ impl QuestDefinition {
                         step.id.clone(),
                     ));
                 }
+
+                for action_item in &task.action_items {
+                    match &*action_item.r#type {
+                        "CUSTOM" => {
+                            if action_item.parameters.get("id").is_none() {
+                                return Err(QuestValidationError::ActionItemParametersNotValid(
+                                    "CUSTOM".to_string(),
+                                ));
+                            }
+                        }
```

Comment:
I think we should not restrict `Custom` type to include an `id` and only check the parameters are not empty.

Reply:
agree!